libcnb-test/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+// Enable rustc and Clippy lints that are disabled by default.
+// https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unused-crate-dependencies
+#![warn(unused_crate_dependencies)]
+// https://rust-lang.github.io/rust-clippy/stable/index.html
+#![warn(clippy::pedantic)]
+// Most of libcnb's public API returns user-provided errors, making error docs redundant.
+#![allow(clippy::missing_errors_doc)]
+// This lint is too noisy and enforces a style that reduces readability in many cases.
+#![allow(clippy::module_name_repetitions)]
+// This lint triggers when both layer_dir and layers_dir are present which are quite common.
+#![allow(clippy::similar_names)]
+#![allow(clippy::default_trait_access)]
+
+mod app;
+mod build;
+mod container_context;
+mod container_port_mapping;
+mod pack;
+mod util;
+
+pub use crate::container_context::{ContainerContext, ContainerExecResult};
```

Comment:
They do use them. `start_container` passes in a `ContainerContext` and `shell_exec` returns a `ContainerExecResult`.

Reply:
Yeah they interact with those structs, but they don't need to import them. If the `pub` is removed here, all tests still pass.

Reply:
But it would omit them from the docs on https://docs.rs/, both the index and they become "unclickable" in the documentation for `start_container` for example.

Reply:
Ah, then that's the use-case :-)